lib/utils/job-utils/http-tool.js

```diff
@@ -0,0 +1,202 @@
+            if (typeof (settings.url) === 'string') {
+                urlObject = urlTool.parse(settings.url);
+            }
+            else {
+                urlObject = settings.url;
+            }
+            // set the REST options
+            urlObject.method = settings.method;
+        
+            // set up the REST headers
+            if (! _.isEmpty(settings.headers)){
+                urlObject.headers = settings.headers;
+            }
+            else {
+                urlObject.headers = {};
+            }
+            
+            urlObject.headers['Content-Length'] = 0;
+            if (settings.hasOwnProperty('data')){
+                this.dataToWrite = JSON.stringify(settings.data);
+                urlObject.headers['Content-Length'] = Buffer.byteLength(this.dataToWrite);
+            }
+            
+            if (! _.isEmpty(settings.credential)){
+                urlObject.auth = settings.credential.username +':'+
+                    settings.credential.password;
+            }
+            
+            // set the protolcol paramter
+            if (urlObject.protocol.substr(-1) !== ':') {
+                urlObject.protocol = urlObject.protocol + ':';
+            }
+
+            urlObject.rejectUnauthorized = settings.verifySSL || false;
+            urlObject.recvTimeoutMs = settings.recvTimeoutMs;
+
+            return urlObject;
+        };
+    }
+    
+    /**
+    * Set up the Request
+    * @param {object} settings - the object that provides the infomation 
+    * of all the works, example:
+    * {
+    *   url: "https://somewhat.website.com:some-port/some/path/to/file",
+    *   method: "GET"/"PUT"/"POST"/"DELETE"/"PATCH",
+    *   credential: {username: "foo", password: "bar"},
+    *   headers: {"token":"whatever-cookie-file"},
+    *   data: "Hello world",
+    *   verifySSL: false,
+    *   recvTimeoutMs: 2000
+    * }
+    * OR
+    * {
+    *   url: {
+    *       protocol: "https",
+    *       host: "somewhat.website.com",
+    *       port: "some-port",
+    *       path: "/some/path/to/file"
+    *   },
+    *   method: "GET"/"PUT"/"POST"/"DELETE"/"PATCH",
+    *   credential: {username: "foo", password: "bar"},
+    *   headers: {"token":"whatever-cookie-file"},
+    *   data: "Hello world",
+    *   verifySSL: false,
+    *   recvTimeoutMs: 2000
+    * }
+    *
+    * @return {Promise}
+    *   resolve() - do whatever next
+    *   reject(err) - do whatever you want with the err
+    */
+
+    HttpTool.prototype.setupRequest = function(settings) {
+        var self = this;
+        var err;
+        return new Promise(function(resolve, reject) {
+            if (! self.isSettingValid(settings)) {
+                err = new Error('Please provide at least url and'+
+                                     ' method to use HTTP tool!');
+                reject(err);
+            }
+            else {
+                if (_.indexOf(self.validMethods, settings.method) > -1) {
```

Comment:
The validation of `method` should belongs to the scope of function `isSettingValid`